packages/docusaurus-plugin-content-docs/src/docs.ts

```diff
@@ -155,8 +155,18 @@ function doProcessDocMetadata({
     // (01-MyFolder/01-MyDoc.md => MyFolder/MyDoc)
     // but allow to disable this behavior with front matter
     parse_number_prefixes: parseNumberPrefixes = true,
+    last_update: lastUpdateFrontMatter,
   } = frontMatter;
 
+  if (lastUpdateFrontMatter?.author && options.showLastUpdateAuthor) {
+    lastUpdate.lastUpdatedBy = lastUpdateFrontMatter.author;
+  }
+  if (lastUpdateFrontMatter?.date && options.showLastUpdateTime) {
+    lastUpdate.lastUpdatedAt =
+      new Date(lastUpdateFrontMatter.date).getTime() / 1000;
+    lastUpdate.formattedLastUpdatedAt = undefined;
+  }
```

Comment:
This can be moved to `readLastUpdateData` so we are clear what the precedence is, and also avoid mutating the object.

Reply:
Currently ```readLastUpdateData``` is called when generating a DocFile to pass into ```processDocMetadata```, where the front matter is parsed. In order to have access to the front matter available when ```readLastUpdateData``` is called, my idea is to move the call inside ```processDocMetadata```. This feels a bit hacky since I would need to have an empty placeholder object for ```lastUpdate``` in the DocFile when passing it into ```processDocMetadata```. Is there a better solution?

Reply:
Ah, I realized when `readLastUpdateData` is called, there's no front matter available yet. In this case, let's defer `readLastUpdateData` to `processDocMetadata` as well, shall we?

Reply:
Will do!